### DISCUSSION TITLE
Please make Inspecor controls available outside of the Inpector

### DISCUSSION THREAD

**Discussion by blackears** [godotengine/godot-proposals#8936] [GUI]
I am finding that I am often creating my own editors for parts of my programs - usually panels registered as plugin docks in the Godot Editor, but sometimes stand alone programs too.  I find myself often wanting to recreate the controls that appear in the inspector inside my panel component.  

The main Controls listed in the library are fairly limited.  It would be nice to have out-of-the-box Components that handle floating point input values (and provide sliders for adjusting the value), Vector3s, editable curves and all the other nifty things that I can see in the Inspector panel.  I find myself continually trying to make crude approximations of these using the available Components.  Since these are already part of the Godot project, it would be really nice if they could be made available for general programs and not just exist in the Inspector.

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/issues/7010 and/or https://github.com/godotengine/godot-proposals/issues/8908.
